### PR TITLE
fixed up standart library imports part 1 of 2

### PR DESCRIPTION
**Pull request by Thextan**
all but the pandas/tests files in this draft PR

### CODE REVIEW

File: pandas/_testing/_io.py [hunk 1]
```diff
@@ -71,10 +72,9 @@
 
 
 def _get_default_network_errors():
-    # Lazy import for http.client & urllib.error
```
Comment (jbrockmendel) [inline pandas/_testing/_io.py:74]: this is imported at runtime for a reason, explained by this comment

File: pandas/core/dtypes/inference.py [hunk 2]
```diff
@@ -417,8 +418,6 @@ def is_dataclass(item):
 
     """
     try:
-        import dataclasses
```
Comment (jbrockmendel) [inline pandas/core/dtypes/inference.py:420]: this is in a try/except for ImportError.  is the ImportError no longer possible?

File: pandas/io/sql.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 )
 from functools import partial
 import re
+import sqlite3
```
Comment (jbrockmendel) [inline pandas/io/sql.py:20]: this adds about 1% to import time

File: pandas/_testing/_io.py (hunk 1 above)
Comment (Thextan) [reply pandas/_testing/_io.py:74]: @jbrockmendel this draft pull was to all @MarcoGorelli to assist me with a specific issue concerning clipboard.
I will revert this change before I submit the real PR.  Thank you for pointing this out.

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import importlib
+from importlib.metadata import entry_points
```
Comment (jbrockmendel) [inline pandas/plotting/_core.py:4]: this add 30ish ms to import time

File: pandas/io/sql.py (hunk 3 above)
Comment (MarcoGorelli) [reply pandas/io/sql.py:20]: thanks @jbrockmendel ! yeah some of these will need adding to the blocklist, I'll take a closer look later

File: pandas/core/dtypes/inference.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/core/dtypes/inference.py:420]: that's right, we're Python3.8+ now. @Thextan could you remove the `except` branch too please?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @Thextan for working on this

The `BLOCKLIST` wasn't large enough, I've made some suggestions to expand it. Could you add those items to it please (preferably alphabetically sorted), and put back the changes to the imports in the blocklist (e.g. `xml` ones)?

**Comment by Thextan** [comment]
@MarcoGorelli I will work on these changes on Tuesday/Wednesday and reach out to you if I need any direction. 
Did you get a chance to look at the issue with pandas/io/clipboard/__init__.py? We discussed it on Wednesday on Slack #newcomers.  Thank you.

File: pandas/io/clipboard/__init__.py [hunk 5]
```diff
@@ -322,19 +335,6 @@ def __setattr__(self, key, value):
 def init_windows_clipboard():
     global HGLOBAL, LPVOID, DWORD, LPCSTR, INT
     global HWND, HINSTANCE, HMENU, BOOL, UINT, HANDLE
-    from ctypes.wintypes import (
```
Comment (MarcoGorelli) [inline pandas/io/clipboard/__init__.py:325]: let's add `ctypes` to the `BLOCKLIST` and revert this change

File: pandas/io/formats/xml.py [hunk 6]
```diff
@@ -9,6 +9,13 @@
     TYPE_CHECKING,
     Any,
 )
+from xml.dom.minidom import parseString
```
Comment (MarcoGorelli) [inline pandas/io/formats/xml.py:12]: let's add `xml` to the BLOCKLIST too

File: pandas/plotting/_core.py (hunk 4 above)
Comment (MarcoGorelli) [reply pandas/plotting/_core.py:4]: @Thextan let's add `importlib.metadata` to the BLOCKLIST as well

File: pandas/_testing/_io.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/_testing/_io.py:74]: let's add `urllib.error` to `BLOCKLIST`

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
yup - I think you can just add that to the BLOCKLIST and revert that change

When this is done we'll also have to check that the timing of `import pandas` isn't affected

File: pandas/_testing/_io.py (hunk 1 above)
Comment (Thextan) [reply pandas/_testing/_io.py:74]: Reverted code back.

File: pandas/io/sql.py (hunk 3 above)
Comment (Thextan) [reply pandas/io/sql.py:20]: Reverted code back.

File: pandas/plotting/_core.py (hunk 4 above)
Comment (Thextan) [reply pandas/plotting/_core.py:4]: Reverted code back.

File: pandas/io/clipboard/__init__.py (hunk 5 above)
Comment (Thextan) [reply pandas/io/clipboard/__init__.py:325]: Added ctypes to BLOCKLIST in scripts/standard_library_imports_should_be_global.py

File: pandas/_testing/_io.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/_testing/_io.py:74]: are you sure? doesn't look like it's reverted

File: pandas/io/clipboard/__init__.py (hunk 5 above)
Comment (MarcoGorelli) [reply pandas/io/clipboard/__init__.py:325]: can you revert the change too please?

File: pandas/core/dtypes/inference.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/core/dtypes/inference.py:420]: can you address this please?

File: pandas/io/formats/xml.py (hunk 6 above)
Comment (MarcoGorelli) [reply pandas/io/formats/xml.py:12]: can you revert the change too please?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by Thextan** [comment]
closing draft pull request.
